README.md

```diff
@@ -288,6 +288,12 @@ const enum Enum { Two } // eslint-disable-line no-const-enum
 You can still disable rules with an .eslintrc.json, but should not in new packages.
 Disabling rules for the entire package makes it harder to review.
 
+##### Formatting with git hooks
+
+Strict formatting is not enforced by default, but you may wish to enable a git hook which will format your code automatically. Run `pnpm run setup-hooks`. Then, when you commit, `dprint fmt` command will be executed on changed declaration files.
+
+If you take advantage of [partial clone](#partial-clone), make sure to call `git sparse-checkout add .husky` to check out the git hooks.
```

Comment:
This is fine, though I personally think we should drop the partial clone instructions altogether from the repo...

Reply:
Curious why you think co. It is really useful to only have a small subset of all files checked out. It takes literal minutes to install everything on the full repo vs seconds when I `sparse-checkout`

Reply:
You may have missed the instructions on how to use pnpm's `--filter` flag; you don't need to install the entire repo, just what you're working on.

The readme is unfortunately huge (and we're adding to it again with this PR), so sorry if it's not clear enough!

Reply:
Interesting. I may have missed, or ignored that indeed.

FWIW, I may still prefer sparse checkout because it's more streamlined. Given a typical workflow, I would checkout the needed subdirectories once and then just run `pnpm i` when I need to. Just like I would with any other repo. The alternative is that I **need to remember** to add the right `--filter` argument, lest I accidentally install everything. This method seems to require more cognitive load.

Not to mention that materialising all files, and there are many, will trigger a massive indexing job. This has always been a PITA with the old setup. My laptop is a lot cooler with sparse checkout :). (I use IntelliJ, don't know if this is relevant to VSCode)

Reply:
I think we largely expect that people will only open the folder of the package they are working on, avoiding that huge indexing problem. vscode has a problem when opening the entire dir (which i haven't had time to investigate).